test/testdrive/cc_cluster_sizes.td

```diff
@@ -50,3 +50,25 @@ contains:unknown cluster replica size 1cc
 
 ! CREATE CLUSTER c SIZE '1C';
 contains:unknown cluster replica size 1C
+
+# Ensure that disk isn't configurable for the new sizes (as it's force enabled).
```

Comment:
We need a test for `ALERT CLUSTER`, too. I think it's possible to create a cluster with the old sizes and disk and then change it to use the new size.

Reply:
Tried this:
```diff
diff --git a/test/testdrive/cc_cluster_sizes.td b/test/testdrive/cc_cluster_sizes.td
index 5e14481269..7ffe8e02e1 100644
--- a/test/testdrive/cc_cluster_sizes.td
+++ b/test/testdrive/cc_cluster_sizes.td
@@ -72,3 +72,10 @@ contains:DISK option not supported for cluster sizes ending in cc or C

 ! CREATE CLUSTER REPLICA c.r SIZE '1cc', DISK = false;
 contains:DISK option not supported for cluster sizes ending in cc or C
+
+> DROP CLUSTER c;
+
+> CREATE CLUSTER c SIZE '1', DISK;
+
+! ALTER CLUSTER c SET (SIZE '1cc');
+contains:DISK option not supported for cluster sizes ending in cc or C
```

It fails with an unexpected error:
```
> ALTER CLUSTER c SET (SIZE '1cc');
query error didn't match; sleeping to see if dataflow produces error shortly 50ms 75ms 113ms 169ms 253ms 380ms 570ms 854ms 1s 2s 3s 4s 6s 10s 15s 22s 33s 22s
^^^ +++
cc_cluster_sizes.td:80:1: error: expected error containing "DISK option not supported for cluster sizes ending in cc or C", got "unknown cluster replica size 1cc"
     |
  10 | $ postgres-execute c ... [rest of line truncated for security]
  25 | $ postgres-execute c ... [rest of line truncated for security]
  45 | $ postgres-execute c ... [rest of line truncated for security]
  79 |
  80 | ! ALTER CLUSTER c SET (SIZE '1cc');
     | ^
```

Let me know if I can help shepherd the PR thorough CI.

Reply:
Ah, great callout! Updated to handle `ALTER` correctly, with a barrage of new test cases. I opted to support altering a cluster that was initially created with an explicitly configured disk...

```
CREATE CLUSTER c (SIZE '1', DISK = {false | true});
ALTER CLUSTER c SET (SIZE '1cc');
```

...since that seems important to facilitate migrations. These scenarios however are disallowed:

```
CREATE CLUSTER c (SIZE '1cc');
ALTER CLUSTER c SET (DISK true);
```

```
CREATE CLUSTER c (SIZE '1');
ALTER CLUSTER c SET (SIZE '1cc', DISK true);
```